src/OpenSearch/Namespaces/AbstractNamespace.php

```diff
@@ -56,16 +68,23 @@ public function extractArgument(array &$params, string $arg)
         }
     }
 
-    protected function performRequest(AbstractEndpoint $endpoint)
+    /**
+     * Perform the request.
+     *
+     * @throws \Psr\Http\Client\ClientExceptionInterface|\Exception
+     */
+    protected function performRequest(EndpointInterface $endpoint): Promise|ResponseInterface
```

Comment:
This return type check actually changes the return type of all public methods on namespaces, which have not been updated